src/pymovements/dataset/dataset_library.py

```diff
@@ -30,55 +36,114 @@ class DatasetLibrary:
 
     Attributes
     ----------
-    definitions: dict[str, type[DatasetDefinition]]
-        Dictionary of :py:class:`~pymovements.dataset.DatasetDefinition`.
+    definitions: dict[str, DatasetDefinition]
+        Dictionary of :py:class:`~pymovements.dataset.DatasetDefinition`,
+        either as classes or instances.
     """
 
-    definitions: dict[str, type[DatasetDefinition]] = {}
+    definitions: dict[str, DatasetDefinition] = {}
 
     @classmethod
-    def add(cls, definition: type[DatasetDefinition]) -> None:
+    def add(cls, definition: type[DatasetDefinition] | DatasetDefinition | Path | str) -> None:
         """Add :py:class:`~pymovements.dataset.DatasetDefinition` to library.
 
         Parameters
         ----------
-        definition: type[DatasetDefinition]
+        definition: type[DatasetDefinition] | DatasetDefinition | Path | str
             The :py:class:`~pymovements.dataset.DatasetDefinition` to add to the library.
+
+        Notes
+        -----
+        Definition can be:
+            * A DatasetDefinition class
+            * A DatasetDefinition instance
+            * A Path to a YAML file
+            * A string path to a YAML file
         """
-        cls.definitions[definition.name] = definition
+        if isinstance(definition, (str, Path)):
+            # Load from YAML file
+            yaml_def = DatasetDefinition.from_yaml(definition)
+            cls.definitions[yaml_def.name] = yaml_def
+        elif isinstance(definition, type):
+            cls.definitions[definition.name] = definition()
+        else:
+            # DatasetDefinition instance
+            cls.definitions[definition.name] = definition
 
     @classmethod
-    def get(cls, name: str) -> type[DatasetDefinition]:
+    def get(cls, name: str) -> DatasetDefinition:
```

Comment:
Yes I also think this change is necessary :+1:

I added this to a breaking changes list at the top of the OP.